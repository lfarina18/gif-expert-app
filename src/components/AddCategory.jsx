import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const onSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length <= 1) return;

    // setCategories((categories) => [inputValue, ...categories]);
    onNewCategory(inputValue.trim());
    setInputValue('');
  };

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired, 
}