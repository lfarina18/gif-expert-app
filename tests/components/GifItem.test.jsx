import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas sobre <GifItem />', () => {
  const title = 'DragonBall';
  const url = `https://api.giphy.com/v1/gifs/search?api_key=LBVMB99LWGBQeagYo9tSKUyPYKTSiBg1&q=${title}&limit=10`;

  test('Debe de hacer match con el snapshot', () => {
    const { container } = render(<GifItem url={url} title={title} />);

    expect(container).toMatchSnapshot();
  });

  test('Debe de mostrar la imagen con el URL y el alt indicado', () => {
    render(<GifItem url={url} title={title} />);

    // expect(screen.getByRole('img').src).toBe(url);

    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('Debe de mostrar el titulo con el componente', () => {
    render(<GifItem url={url} title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
