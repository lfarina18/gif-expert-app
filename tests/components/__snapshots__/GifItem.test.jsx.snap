// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GifItem /> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="DragonBall"
      src="https://api.giphy.com/v1/gifs/search?api_key=LBVMB99LWGBQeagYo9tSKUyPYKTSiBg1&q=DragonBall&limit=10"
    />
    <p>
      DragonBall
    </p>
  </div>
</div>
`;
